web_external/js/views/StudyResults/StudyResultsLocalFeaturesView.js

```diff
@@ -0,0 +1,145 @@
+            hasLocalFeatures: !_.isEmpty(this.featureset.get('localFeatures'))
+        }));
+
+        this.selectFeatureView = new isic.views.StudyResultsLocalFeaturesSelectView({
+            el: this.$('#isic-study-results-local-features-select-container'),
+            annotation: this.annotation,
+            featureset: this.featureset,
+            parentView: this
+        });
+        this.overlayImageViewerWidget = new isic.views.OverlayImageViewerWidget({
+            el: this.$('#isic-study-results-local-features-image-container'),
+            model: this.annotation.image(),
+            parentView: this
+        }).render();
+
+        this.selectedFeatureId = null;
+        this.listenTo(this.selectFeatureView, 'changed', this.selectedFeatureChanged);
+
+        return this;
+    },
+
+    selectedFeatureChanged: function (featureId) {
+        if (featureId === this.selectedFeatureId) {
+            return;
+        }
+        this.selectedFeatureId = featureId;
+
+        if (this.selectedFeatureId && this.annotation.isComplete()) {
+            var annotationValue = this.annotation.get('annotations')[this.selectedFeatureId];
+            if (annotationValue) {
+                this.overlayImageViewerWidget.overlay(annotationValue);
+                return;
+            }
+        }
+        this.overlayImageViewerWidget.clear();
+    },
+
+    annotationChanged: function () {
+        this.selectedFeatureId = null;
+
+        // this.selectFeatureView updates itself when this.annotation changes
+
+        // TODO: this.overlayImageViewerWidget should probably expose a method
+        //   to update its model
+        var overlayImageModel = this.overlayImageViewerWidget.model;
+        overlayImageModel.clear({silent: true});
+        overlayImageModel.set(this.annotation.image().attributes);
+    },
+
+    setVisible: function (visible) {
+        if (visible) {
+            this.$el.removeClass('hidden');
+        } else {
+            this.$el.addClass('hidden');
+        }
+    }
+});
+
+// View for a collection of local features in a select tag
+isic.views.StudyResultsLocalFeaturesSelectView = isic.View.extend({
+    events: {
+        'change': 'selectedFeatureChanged'
+    },
+
+    initialize: function (settings) {
+        this.annotation = settings.annotation;
+        this.featureset = settings.featureset;
+
+        this.render();
+
+        this.listenTo(this.annotation, 'change', this.annotationChanged);
+    },
+
+    render: function () {
+        var availableFeatures;
+        if (this.annotation.isComplete()) {
+            // Annotation is complete
+            var featuresetLocalFeatures = this.featureset.get('localFeatures');
+            var annotationValues = this.annotation.get('annotations');
+            availableFeatures = featuresetLocalFeatures.filter(_.bind(function (feature) {
+                return _.has(annotationValues, feature.id);
```

Comment:
I see errors here when selecting users, like:

    underscore.js:1046 Uncaught TypeError: Cannot convert undefined or null to object(…)
        x.has @ underscore.js:1046
        (anonymous function) @ StudyResultsLocalFeaturesView.js:111